src/cc65/declare.c

```diff
@@ -840,10 +841,13 @@ static SymEntry* ParseStructDecl (const char* Name)
                 AddBitField (Decl.Ident, Offs, BitOffs % CHAR_BITS, FieldWidth);
                 BitOffs += FieldWidth;
                 CHECK (BitOffs <= (int) INT_BITS);
-                if (BitOffs == INT_BITS) {
-                    StructSize += SIZEOF_INT;
-                    BitOffs = 0;
-                }
+                /* The code above should either be rewritten to use these
+                ** values, or overlap of bit-field storage units should be
+                ** made into an option or controlled by something like
+                ** __attribute__((__packed__)).
+                */
+                StructSize += BitOffs / CHAR_BITS;
+                BitOffs %= CHAR_BITS;
```

Comment:
I just wrote it like this to make the diff clearer. It could also be combined with the code above, but I'm wondering if the old and new behavior should be selectable by a command line flag or with `__atttribute__` or similar.

Reply:
The 6502 doesn't benefit from alignment. Consequently cc65 always "packs" structs. In other words: Its struct layout is by default the same as with compilers for "usual" targets when called with `-fpack-struct`. From that perspective I don't see any point in some option here. cc65 should here as well just behave like compilers for "usual" targets when called with `-fpack-struct` - that's by far the most anticipation-compliant behavior. So please write the code in the way that makes most sense with that in mind.

Reply:
Updated.